firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -4537,31 +4536,76 @@ public void onComplete(DatabaseError error, DatabaseReference ref) {
     IntegrationTestHelpers.waitFor(semaphore);
   }
 
-  private static FirebaseApp appForDatabaseUrl(String url, String name) {
-    return FirebaseApp.initializeApp(
-        InstrumentationRegistry.getInstrumentation().getTargetContext(),
-        new FirebaseOptions.Builder()
-            .setApplicationId("appid")
-            .setApiKey("apikey")
-            .setDatabaseUrl(url)
-            .build(),
-        name);
+  @Test
+  public void testGetPropagatesEventsToListeners()
+      throws DatabaseException, InterruptedException, ExecutionException {
+    /**
+     * There's a situation where if you have a pending `get()` and then an `onValue()` at the same
+     * location and get resolves before onValue does, onValue gets no events. Note: This is a smoke
+     * test due to the fact that it is difficult to reproduce the scenario when the parent get()
+     * request resolves before a child onValue listener.
+     */
+    // To ensure that we don't read the cached results, we need a separate app.
+    AtomicBoolean successful = new AtomicBoolean();
+    int count = 0;
+    while (count++ < 5 && !successful.get()) {
+
+      FirebaseDatabase readDb = IntegrationTestHelpers.getNewDatabase();
+      FirebaseDatabase writeDb = IntegrationTestHelpers.getNewDatabase();
```

Comment:
We're using the unqualified `getNewDatabase()` elsewhere:

```
FirebaseDatabase writeDb = getNewDatabase();
```

We can be consistent and drop `IntegrationTestHelpers`.

Reply:
I'd say for all the commonly used helpers in `IntegrationTestHelpers` let's use the unqualified form.